### PR TITLE
blockcopy:Ignore non-disk disks

### PR DESCRIPTION
**Pull request by chloerh**
In case making invalid snapshot that has not 'source' attrib

Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/virsh_cmd/domain/virsh_blockcopy.py [hunk 1]
```diff
@@ -367,6 +367,10 @@ def _make_snapshot():
         # Add all disks into xml file.
         vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
         disks = vmxml.devices.by_device_tag('disk')
+        # Remove non-storage disk such as 'cdrom'
+        for dis in disks:
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/domain/virsh_blockcopy.py]: let us have the name complete
`for disk in disks:`

File: libvirt/tests/src/virsh_cmd/domain/virsh_blockcopy.py [hunk 2]
```diff
@@ -367,6 +367,10 @@ def _make_snapshot():
         # Add all disks into xml file.
         vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
         disks = vmxml.devices.by_device_tag('disk')
+        # Remove non-storage disk such as 'cdrom'
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_blockcopy.py]: Beside cdrom, how about floppy? Do we need remove it too?

Comment (chloerh) [reply libvirt/tests/src/virsh_cmd/domain/virsh_blockcopy.py]: Will remove all disks that 'device' is not 'disk', no matter it's 'cdrom' or 'floppy'

**Comment by chunfuwen** [comment]
@chloerh ,please reply balamuruhans's comment about dis in disks:

File: libvirt/tests/src/virsh_cmd/domain/virsh_blockcopy.py (hunk 1 above)
Comment (chloerh) [reply libvirt/tests/src/virsh_cmd/domain/virsh_blockcopy.py]: Updated, thanks for the review

**Comment by balamuruhans** [comment]
Thanks @chloerh @chunfuwen

**Pull request merged**